src/lumigo_tracer/lambda_tracer/lambda_reporter.py

```diff
@@ -163,33 +166,62 @@ def report_json(
             if not edge_connection:
                 get_logger().warning("Can not establish connection. Skip sending span.")
                 return duration
+
+    lock = threading.Lock()
+
+    def send_single_span(data: str, retry: bool = True) -> None:
+        """
+        Helper function to send a single span request and handle retries,
+        including re-establishing connection if necessary.
+        """
+        global edge_connection
```

Comment:
Why didn't you pass the edge connection as input to the thread function?

`global` is messy, use it only if necessary

Reply:
Changed

Reply:
Changed